src/app/pages/candidate-details/__tests__/candidate-details.page.spec.ts

```diff
@@ -136,10 +139,31 @@ describe('CandidateDetailsPage', () => {
 
   describe('dismiss', () => {
     it('should dismiss open modal', fakeAsync(async () => {
-      spyOn(component.modalController, 'dismiss');
-      await component.dismiss();
-      expect(component.modalController.dismiss).toHaveBeenCalled();
+      spyOn(component.modalController, 'dismiss').and.callThrough();
+      spyOn(routerSpy, 'url').and.returnValue('/test');
+      spyOn(component, 'formatUrl').and.returnValue('test');
+      spyOn(store$, 'dispatch').and.stub();
```

Comment:
When we spyOn all parts of a method we don't really test it as we are controlling all of the inner workings. This test should be as simple as 

```
  describe('dismiss', () => {
    it('should dismiss open modal', fakeAsync(async () => {
      spyOn(store$, 'dispatch');
      spyOn(component.modalController, 'dismiss').and.returnValue(Promise.resolve(true));
      await component.dismiss();
      expect(component.modalController.dismiss).toHaveBeenCalled();
      expect(store$.dispatch).toHaveBeenCalledWith(candidateDetailActions.CandidateDetailsModalDismiss(
        { sourcePage: 'url' },
      ));
    }));
  });
```